app/knights/knight.py

```diff
@@ -0,0 +1,35 @@
+class Knight:
+    def __init__(
+            self,
+            name: str,
+            power: int,
+            hp: int,
+            armour: list,
+            weapon: dict,
+            potion: None | dict,
+            protection: int = 0
+    ) -> None:
+        self.name = name
+        self.power = power
+        self.hp = hp
+        self.armour = armour
+        self.weapon = weapon
+        self.potion = potion
+        self.protection = protection
+
+    def apply_armour(self) -> None:
+        for armour in self.armour:
+            self.protection += armour["protection"]
+
+    def apply_weapon(self) -> None:
+        self.power += self.weapon["power"]
+
+    def apply_potion(self) -> None:
+        if self.potion is not None:
+            effect = self.potion["effect"]
+            if "power" in effect:
+                self.power += effect["power"]
+            if "protection" in effect:
+                self.protection += effect["protection"]
+            if "hp" in effect:
+                self.hp += effect["hp"]

```

Comment:
Use loop with getattr and setattr

Reply:
We haven't passed it yet. Do you have to figure it out yourself or leave it like that?

Reply:
Yes, you have to learn something yourself. Improve this please